### PR TITLE
Add tutorial on integrating kubo with index-provider

### PR DESCRIPTION
**Pull request by ischasny**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -117,6 +117,116 @@ Delegated Routing server is off by default. To enable it, add the following conf
 }
 ```
 
+#### Configuring Kubo and index-provider to use IPNI
+
+Kubo supports HTTP delegated routing as of v0.20.0. The following section contains configuration examples and a few tips to enable Kubo to advertise its CIDs to 
+IPNI systems like `cid.contact` using `index-provider`. Delegated Routing is still in the Experimental stage and configuration might change from version to version. 
+This section is not Alma Mater for all-things Kubo. It rather exists to give one inspiration of how to configure their node to use IPNI. 
+Please refer to the [Kubo docs](https://docs.ipfs.tech/install/command-line/) for more thorough info. A few things to keep in mind:
```
Comment (masih) [inline README.md:125]:
```suggestion
This section serves as an inspiration for configuring your node to use IPNI, but for comprehensive information, refer to the [Kubo documentation](https://docs.ipfs.tech/install/command-line/). Here are some important points to consider:
```

File: README.md [hunk 2]
```diff
@@ -117,6 +117,116 @@ Delegated Routing server is off by default. To enable it, add the following conf
...
+
+* `index-provider`'s delegated routing server should be run at all times as a "sidecar" to the Kubo node. `index-provider` can be safely restarted, however
+if it's down, no new CIDs will be flowing from Kubo to IPNI;
+* `index-provider` doesn't support Reframe anymore. Please upgrade to Kubo v0.20.+ with HTTP Delegated Routing support;
+* Kubo advertises it's data in _snapshots_. That means that all CIDs under its management get _reprovided_ into the configured routers every 12/24 hours (configurable).
+That mechanism takes its roots from the way DHT works. `index-provider` can track what CIDs have been removed / added in between two consecuitive snapshots and convert
+that information into IPNI advertisements. Depending on the size of the node, during reproviding process one might see a lot of chatter between all processes involved.
+In between reprovides, Kubo still sends new individual CIDs to the configured routers;
+* Kubo needs `index-provider` only to publish its CIDs to IPNI. Kubo can do IPNI lookups natively without a sidecar involved (see the Kubo docs on `auto` routers);
```
Comment (masih) [inline README.md:134]:
```suggestion
* The `index-provider` delegated routing server should be running continuously as a "sidecar" to the Kubo node. While `index-provider` can be restarted safely, if it goes down, no new CIDs will flow from Kubo to IPNI.
* The latest version of Kubo (v0.20.+) with HTTP delegated routing support should be used as `index-provider` no longer supports Reframe.
* Kubo advertises its data in snapshots, which means that all CIDs managed by Kubo are reprovided to the configured routers every 12/24 hours (configurable). This mechanism is similar to how the Distributed Hash Table (DHT) works. During the reproviding process, there may be significant communication between the involved processes. In between reprovides, Kubo also sends new individual CIDs to the configured routers.
* Kubo requires `index-provider` only for publishing its CIDs to IPNI. Kubo can perform IPNI lookups natively without the need for a sidecar (refer to Kubo docs on `auto` routers).
```

**Review by masih — APPROVED** [review]
Thank you for adding this @ischasny 🍻

**Comment by ischasny** [comment]
> Hi all, FYSA these docs encourage use of `PUT /routing/v1` feature which has no finalized spec / API ([IPIP-378](https://github.com/ipfs/specs/pull/378) is wip), no official support in Kubo, but we already have confused users fill bugs caused by your use case and docs, see [ipfs/kubo#9823 (comment)](https://github.com/ipfs/kubo/issues/9823#issuecomment-1572198887)
> 
> As this is not ready on the Kubo end, nor we have bandwidth to tackle yet another work stream with the skeleton crew, you need to make a decision: either you contribute time to fix the bug, and all future ones, or remove the docs. 🙏

Hi @lidel. Would you be happy if we explicitly mark it as an :skull_and_crossbones:Unsupported experimental feature, use it at your own risk :skull_and_crossbones: kind of marker? The reason I’m asking is because we have users who used to use reframe so at least we can provide them with some path.

**Comment by ischasny** [comment]
https://github.com/ipni/index-provider/pull/373

**Pull request merged**